linux/LinuxMachine.c

```diff
@@ -674,6 +772,13 @@ Machine* Machine_new(UsersTable* usersTable, uid_t userId) {
    // Initialize CPU count
    LinuxMachine_updateCPUcount(this);
 
+   #ifdef HAVE_SENSORS_SENSORS_H
+   // Fetch CPU topology
+   LinuxMachine_fetchCPUTopologyFromCPUinfo(this);
```

Comment:
maybe `LinuxMachine_fetchCPUTopologyFromCPUinfo()` can return whether the topology changed and then `LibSensors_countCCDs()` and `LinuxMachine_assignCCDs()` get only called on change?

Reply:
I don't know what this means, this code runs only once already?